consensus/spos/bls/subroundSignature.go

```diff
@@ -280,6 +279,8 @@ func (sr *subroundSignature) waitAllSignatures() {
 		return
 	}
 
+	sr.WaitingAllSignaturesTimeOut = true
```

Comment:
can we have a race condition here? If yes, you can use atomic.Flag

Reply:
Actually not. As the reset is done on startRound  initialization and this go routine stays live only 55% of the current round and this set on true is done in the middle of the current round

Reply:
+1

Reply:
But we can use atomic.Flag anyway

Reply:
Not necessarily if we can not have a race condition. Asked just to double-check. Green from me